lib/puppet/provider/service/systemd.rb

```diff
@@ -145,5 +145,36 @@ def stopcmd
   def statuscmd
     [command(:systemctl), "is-active", @resource[:name]]
   end
+
+  def restart
+    begin
+      super
+    rescue Puppet::Error => e
+      raise Puppet::Error.new(prepare_error_message(@resource[:name], 'restart', e))
+    end
+  end
+
+  def start
+    begin
+      super
+    rescue Puppet::Error => e
+      raise Puppet::Error.new(prepare_error_message(@resource[:name], 'start', e))
+    end
+  end
+
+  def stop
+    begin
+      super
+    rescue Puppet::Error => e
+      raise Puppet::Error.new(prepare_error_message(@resource[:name], 'stop', e))
+    end
+  end
+
+  def prepare_error_message(name, action, exception)
+    error_return = "Systemd #{action} for #{name} failed:"
+    journalctl_output = execute("journalctl -n 50 --since '5 minutes ago' -u #{name} --no-pager")
+    error_return << "journalctl log for #{name}: #{journalctl_output}"
+    raise Puppet::Error, error_return, exception.backtrace
```

Comment:
The calling methods are calling this method to generate an error message for an exception, but this code itself raises an exception - which makes the `raise` calls in the start/stop/restart methods dead code.